### PR TITLE
Oss widget update

### PR DESCRIPTION
**Pull request by tatlax3636**
* incorporated dropdown and refresh functionality similar to SAST widget
* moved the refresh-modal to shared since it will be used by all widgets with that functionality

### CODE REVIEW

File: src/app/widget_modules/opensource-scan/opensource-scan.service.spec.ts [hunk 1]
```diff
@@ -15,11 +15,75 @@ describe('OpensourceScanService', () => {
...
+              branch: 'development',
+              url: 'url.com',
+              repoName: 'identitygithub.com',
+              path: '/test',
+              artifactName: 'artifactTest',
+              password: 'pswrd',
+              personalAccessToken: 'token'
+            },
+            pushed: false,
+            refreshLink: 'http://localhost:8081/security/refresh?projectName=test-master',
```
Comment (nameisaravind) [inline src/app/widget_modules/opensource-scan/opensource-scan.service.spec.ts:72]: @tatlax3636 update url with without host

File: src/app/widget_modules/opensource-scan/oss-widget/oss-widget.component.spec.ts [hunk 2]
```diff
@@ -25,10 +26,40 @@ import {OpensourceScanModule} from '../opensource-scan.module';
...
+        branch: 'development',
+        url: 'url.com',
+        repoName: 'identitygithub.com',
+        path: '/test',
+        artifactName: 'artifactTest',
+        password: 'pswrd',
+        personalAccessToken: 'token'
+      },
+      pushed: false,
+      refreshLink: 'http://localhost:8081/security/refresh?projectName=identity-profile-preferences-master',
```
Comment (nameisaravind) [inline src/app/widget_modules/opensource-scan/oss-widget/oss-widget.component.spec.ts:53]: update url without host

File: src/app/widget_modules/opensource-scan/oss-widget/oss-widget.component.spec.ts [hunk 3]
```diff
@@ -67,6 +102,35 @@ describe('OSSWidgetComponent', () => {
...
+          branch: 'development',
+          url: 'url.com',
+          repoName: 'identitygithub.com',
+          path: '/test',
+          artifactName: 'artifactTest',
+          password: 'pswrd',
+          personalAccessToken: 'token'
+        },
+        pushed: false,
+        refreshLink: 'http://localhost:8081/security/refresh?projectName=identity-profile-preferences-master',
```
Comment (nameisaravind) [inline src/app/widget_modules/opensource-scan/oss-widget/oss-widget.component.spec.ts:128]: update url without host

File: src/app/widget_modules/opensource-scan/oss-widget/oss-widget.component.ts [hunk 4]
```diff
@@ -109,10 +126,35 @@ export class OSSWidgetComponent extends WidgetComponent implements OnInit, After
...
+        this.setDefaultIfNoData();
+      }
+
+    });
+
+  }
+
+  populateDropdown(collectorItems) {
+    collectorItems.map(item => {
+      item.description = item.description.split(':')[0];
```
Comment (nameisaravind) [inline src/app/widget_modules/opensource-scan/oss-widget/oss-widget.component.ts:155]: add npe for description

File: src/app/widget_modules/opensource-scan/opensource-scan.service.spec.ts (hunk 1 above)
Comment (tatlax3636) [reply src/app/widget_modules/opensource-scan/opensource-scan.service.spec.ts]: Updated

File: src/app/widget_modules/opensource-scan/oss-widget/oss-widget.component.spec.ts (hunk 2 above)
Comment (tatlax3636) [reply src/app/widget_modules/opensource-scan/oss-widget/oss-widget.component.spec.ts]: updated

File: src/app/widget_modules/opensource-scan/oss-widget/oss-widget.component.spec.ts (hunk 3 above)
Comment (tatlax3636) [reply src/app/widget_modules/opensource-scan/oss-widget/oss-widget.component.spec.ts]: updated

File: src/app/widget_modules/opensource-scan/oss-widget/oss-widget.component.ts (hunk 4 above)
Comment (tatlax3636) [reply src/app/widget_modules/opensource-scan/oss-widget/oss-widget.component.ts]: updated

**Review by nameisaravind — APPROVED** [review]
lgtm

**Pull request merged**